main.rb

```diff
@@ -1,43 +1,212 @@
-puts "#{lily.name} #{lily.classroom.label}"
+def prompt_create_rental
+  puts ''
+  print 'SELECT BOOK TO RENT:> '
+  book_idx = gets.chomp.to_i
+  person_instances = list_all_people
 
-notw = Book.new('Name of the Wind', 'Patrick Rothfuss')
-sla = Book.new('Stormlight Archives', 'Brandon Sanderson')
+  print 'SELECT RENTER[ID]:> '
+  renter_id = gets.chomp.to_i
+  print 'Date:> '
+  date = gets.chomp
+  [person_instances, book_idx, renter_id, date]
+end
 
-abebe = Person.new(34, 'Abebe', parent_permission: true)
-zeritu = Person.new(50, 'Zeritu', parent_permission: true)
+def create_rental
+  book_instances = list_all_books
+  params = prompt_create_rental
+  puts ''
+  selected_book = nil
+  selected_person = nil
+  book_instances.each_with_index do |book, index|
+    selected_book = book if (params[1] - 1) == index
+  end
 
-rental1 = Rental.new('2021-09-12', notw, abebe)
-rental2 = Rental.new('2022-04-14', sla, abebe)
+  params[0].each_with_index do |person, _index|
+    selected_person = person if params[2] == person.id
+  end
+  Rental.new(params[3], selected_book, selected_person)
+  puts ''
+  puts 'Rental created successfully!'
+  puts ''
+end
 
-rental3 = Rental.new('2023-04-03', sla, zeritu)
-rental4 = Rental.new('2023-07-02', notw, zeritu)
+def handle_case(input)
+  actions = {
+    1 => method(:list_all_books),
+    2 => method(:list_all_people),
+    3 => method(:create_person),
+    4 => method(:create_book),
+    5 => method(:create_rental),
+    6 => method(:handle_rental_list),
+    7 => method(:exit)
+  }
 
-got = Book.new('Game of thrones', 'Goeorge R.R Martin')
+  action = actions[input]
+  action ? action.call : handle_wrong_input
+end
 
-abebe.add_rental('2023-08-24', got)
-got.add_rental('2023-08-24', abebe)
+def handle_rental_list
+  puts ''
+  list_all_people
+  puts ''
+  print 'TYPE RENTER ID:> '
+  id = gets.chomp.to_i
+  list_all_rentals(id)
+end
 
-puts "#{rental1.person.name} => #{rental1.book.title}"
-puts "#{rental2.person.name} => #{rental2.book.title}"
-puts "#{rental3.person.name} => #{rental3.book.title}"
-puts "#{rental4.person.name} #{rental4.book.title}"
+def handle_wrong_input
+  puts 'Wrong input, please enter again!'
+end
 
-puts "#{abebe.name} #{abebe.rentals[0].book.title}"
-puts "#{got.title} #{got.rentals[0].person.name}"
+def main
+  loop do
+    puts ''
+    puts '=================================================='
+    puts '=   PLEASE CHOOSE AN OPTION BY TYPING A NUMBER   ='
+    puts '=================================================='
+    puts ''
+    puts '  1 - List of all books'
+    puts '  2 - List of all people'
+    puts '  3 - Create a person'
+    puts '  4 - Create a book'
+    puts '  5 - Create a rental'
+    puts '  6 - List od all rentals for a given id'
+    puts '  7 - Exit'
+    print '  :> '
+    input = gets.chomp.to_i
+    handle_case(input)
+    puts ''
+    puts '**************************************************'
+  end
+  puts '\n\n'
+end
+main
```

Comment:
> - [ ] Hello @wacoo, I'd like to share some feedback based on the previous reviewer's comments. It appears there might be a slight misunderstanding, and it seems that you've renamed the app.rb file to main.rb. However, this naming adjustment is not acceptable. Instead, it is kindly recommended that you create a separate main.rb file and utilize the example code provided below:
> 
> Please create a new main.rb file and add the following code to it:
> 
> Create main.rb file:
> 
> ```
> require_relative('app')
> 
> def main
>   loop do
>     puts ''
>     puts '=================================================='
>     puts '=   PLEASE CHOOSE AN OPTION BY TYPING A NUMBER   ='
>     puts '=================================================='
>     puts ''
>     puts '  1 - List of all books'
>     puts '  2 - List of all people'
>     puts '  3 - Create a person'
>     puts '  4 - Create a book'
>     puts '  5 - Create a rental'
>     puts '  6 - List od all rentals for a given id'
>     puts '  7 - Exit'
>     print '  :> '
>     input = gets.chomp.to_i
>     handle_case(input)
>     puts ''
>     puts '**************************************************'
>   end
>   puts '\n\n'
> end
> 
> def handle_case(input)
>   actions = {
>     1 => method(:list_all_books),
>     2 => method(:list_all_people),
>     3 => method(:create_person),
>     4 => method(:create_book),
>     5 => method(:create_rental),
>     6 => method(:handle_rental_list),
>     7 => method(:exit)
>   }
> 
>   action = actions[input]
>   action ? action.call : handle_wrong_input
> end
> 
> main
> ```
> 
> After creating the new main.rb file, kindly rename this file back to app.rb, and also, please remove the code starting from line 177 to line 184, which is the code for handle_case(input). This will help you to meet the project requirements. Your cooperation in making these changes would be greatly appreciated. 👍